samples: cellular: nRF Cloud auto-onboarding (#14081)
nrfconnect/sdk-nrf: samples/cellular/nrf_cloud_rest_device_message/README.rst

.. _nrf_cloud_rest_device_message_provisioning:

Provisioning and onboarding your device to nRF Cloud
****************************************************

Your device must be provisioned with nRF Cloud for this sample to function.
You only need to do this once for each device.

> There are two main methods to provision your device:

Reviewer: ```suggestion
This sample supports two methods to on-board your device:
```

1. Using `nRF Cloud Provisioning Service`_ and the auto-onboarding option.
   This is the preferred option.
#. Using Just-in-time provisioning.
   This is the legacy option.

Remote provisioning and auto-onboarding with nRF Cloud
======================================================

Build and run the sample with the remote provisioning overlay as explained below: :ref:`nrf_cloud_rest_device_message_remote_provisioning_overlay`.

> See :ref:`nRF Cloud Auto-onboarding` for more information.

Reviewer: Explicitly state that the user must follow steps outlined in:
```
:ref:`nRF Cloud device claiming`
```

The device ID will be in the UUID format, not the legacy 'nrf-\ *IMEI*\ ' format.

Just-in-time provisioning (JITP) with nRF Cloud
===============================================

> Complete the following steps to provision your device:

Reviewer: ```suggestion
Complete the following steps to on-board your device:
```

.. _nrf_cloud_rest_device_message_provisioning:

Provisioning and onboarding your device to nRF Cloud
****************************************************

You must on-board your device to nRF Cloud for this sample to function.
You only need to do this once for each device.

This sample supports two methods to on-board your device:

> 1. Using `nRF Cloud Provisioning Service`_ and the auto-onboarding option.

Reviewer 2: ```suggestion
* Using `nRF Cloud Provisioning Service`_ and the auto-onboarding option.
```

   This is the preferred option for nRF91x1-based devices.
   The nRF Cloud Provisioning Service auto-onboarding is currently compatible with REST and CoAP but not MQTT connectivity to nRF Cloud; for that, use the next method.

> #. Using Just-in-time provisioning.

Reviewer 2: ```suggestion
* Using Just-in-time provisioning.
```

   This is the legacy option.
   Use this method for nRF9160-based devices.

Remote provisioning and auto-onboarding with nRF Cloud
======================================================

Build and run the sample with the remote provisioning overlay as explained below: :ref:`nrf_cloud_rest_device_message_remote_provisioning_overlay`.
Follow the steps outlined in `device claiming <nRF Cloud device claiming_>`_.
See `nRF Cloud Auto-onboarding`_ for more information.

> The device ID will be in the UUID format, not the legacy 'nrf-\ *IMEI*\ ' format.

Reviewer 2: ```suggestion
The device ID is in the UUID format, not the legacy 'nrf-\ *IMEI*\ ' format.
```

Set the :kconfig:option:`CONFIG_NRF_CLOUD_LOG_OUTPUT_LEVEL` option to the log level of messages to send to nRF Cloud, such as ``4`` for debug log messages.
Set the :kconfig:option:`CONFIG_NRF_CLOUD_REST_DEVICE_MESSAGE_SAMPLE_LOG_LEVEL_DBG` option so that log messages are generated on each button press.

.. _nrf_cloud_rest_device_message_remote_provisioning_overlay:

Remote provisioning overlay
===========================

> 

Reviewer: this section should include a note about the UUID device id format that is now enabled by using this overlay.
